src/model/consent/consent.ts

```diff
@@ -57,16 +66,20 @@ export class ConsentDB {
   }
 
   // Update Consent credential
+  // Only non-null Consent fields are updated
   public updateCredentials (consent: Consent): Promise<Consent[]> {
+    const validatedConsent = {}
+
+    for (const key of Object.keys(consent)) {
```

Comment:
you can use the simpler construct when apply `in` 
```
for (const property in object) {
  console.log(`${property}: ${object[property]}`);
}
```

Reply:
Thank you for the suggestion! Implemented it.